### PR TITLE
PT-668 | Add matomo analytics

### PR DESCRIPTION
**Pull request by Sartonon**
## Description :sparkles:

- Add matomo analytics

## Additional notes :spiral_notepad:

### CODE REVIEW

**Review by nikomakela — APPROVED** [review]
Looks good to me

File: .github/workflows/review.yml [hunk 1]
```diff
@@ -30,6 +30,9 @@ jobs:
           DOCKER_BUILD_ARG_NEXT_PUBLIC_SENTRY_DSN: ${{ secrets.GH_SENTRY_DSN }}
           DOCKER_BUILD_ARG_NEXT_PUBLIC_ENVIRONMENT: 'review'
           DOCKER_BUILD_ARG_NEXT_PUBLIC_API_BASE_URL: 'https://palvelutarjotin-api.test.kuva.hel.ninja/graphql'
+          DOCKER_BUILD_ARG_NEXT_PUBLIC_MATOMO_URL_BASE: 'https://analytics.hel.ninja/'
```
Comment (nenonja) [inline .github/workflows/review.yml:33]: does this pollute production analytics with test env analytics? Test environments should have it's own url?

**Review by nenonja — COMMENTED** [review]
one comment / question

File: src/domain/app/layout/PageWrapper.tsx [hunk 2]
```diff
@@ -12,12 +13,21 @@ interface Props {
...
   const locale = useLocale();
+  const { trackPageView } = useMatomo();
   const { asPath: pathname } = useRouter();
+
   const translatedTitle =
     title !== 'appName'
       ? `${t(title)} - ${t('common:appName')}`
       : t('common:appName');
 
+  // Track page changes when pathnname changes
```
Comment (nenonja) [inline src/domain/app/layout/PageWrapper.tsx:24]: What is the level of analytics we try to achieve? As far as I know, we could track that does users follow certain links, do interactions, how long they stay on the site etc.

Comment (Sartonon) [reply src/domain/app/layout/PageWrapper.tsx:24]: I have no idea what kinds of analytics we want.. I think the task was to get this set up. I haven't investogated in matamo what is it doing by default.

Comment (Sartonon) [reply src/domain/app/layout/PageWrapper.tsx:24]: This page change was basically copied from events-helsinki-ui

File: .github/workflows/review.yml (hunk 1 above)
Comment (Sartonon) [reply .github/workflows/review.yml:33]: It is not activated in staging for example. Maybe we could delete the env variables from envs where they are not really used

File: src/domain/app/layout/PageWrapper.tsx (hunk 2 above)
Comment (nenonja) [reply src/domain/app/layout/PageWrapper.tsx:24]: Requirements for analytics should come from stakeholders

File: .github/workflows/review.yml (hunk 1 above)
Comment (nenonja) [reply .github/workflows/review.yml:33]: So this can be / should be removed + also from staging.yml ?

**Review by nenonja — APPROVED** [review]

**Pull request merged**